staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -332,7 +332,9 @@ func (rules *ClientConfigLoadingRules) GetExplicitFile() string {
 	return rules.ExplicitPath
 }
 
-// IsDefaultConfig returns true if the provided configuration matches the default
+// IsDefaultConfig returns true if the provided configuration matches the default, but
```

Comment:
Can you explain why doing this fixes the bug? My naive take is that doing `reflect.DeepEqual` is a really weird outside of tests, and I can't really reason when or why you'd use a method like `IsDefaultConfig`.

Reply:
this is revealing that the comparison is being done at the wrong point...

currently, this is what is happening:
1. load kubeconfig if specified
2. merge in command line overrides
3. if the result is equal to what you would get if you loaded an empty kubeconfig file, discard it and switch to the in-cluster-config

The comparison should be done after step 1, then apply command line overrides. This is a more fundamental (but correct) change

Reply:
> The comparison should be done after step 1, then apply command line overrides.

Hi, I tried your suggestion.
If we change the order of comparison steps, It will not make thinks less complex, I think.
For example, if user specify --namespace=foo, will we apply namespace foo to in-cluster config?